src/iterators/linear.rs

```diff
@@ -6,20 +6,23 @@ use iterators::{HistogramIterator, PickyIterator};
 pub struct Iter<'a, T: 'a + Counter> {
     hist: &'a Histogram<T>,
 
-    valueUnitsPerBucket: i64,
-    currentStepHighestValueReportingLevel: i64,
-    currentStepLowestValueReportingLevel: i64,
+    // > 0
+    valueUnitsPerBucket: u64,
+    currentStepHighestValueReportingLevel: u64,
+    currentStepLowestValueReportingLevel: u64,
 }
 
 impl<'a, T: 'a + Counter> Iter<'a, T> {
     /// Construct a new linear iterator. See `Histogram::iter_linear` for details.
     pub fn new(hist: &'a Histogram<T>,
-               valueUnitsPerBucket: i64)
+               valueUnitsPerBucket: u64)
                -> HistogramIterator<'a, T, Iter<'a, T>> {
+        assert!(valueUnitsPerBucket > 0);
```

Comment:
No, I'm happy to leave them as `assert!`. *Maybe* they should be `debug_assert!` instead though? What do you think?